File: 프로그래머스/0/120583. 중복된 숫자 개수/중복된 숫자 개수.java
import java.util.*;

class Solution {
    public int solution(int[] array, int n) {
        int answer = 0;
        
        for(int i : array) {
            if(i == n) {
                answer++;
            }
        }
        
        return answer;
    }
}

// 다른 풀이
 return (int) Arrays.stream(array).filter(e -> e == n).count();
// .filter() : 스트림에서 주어진 조건을 만족하는 요소만을 선택하여 새로운 스트림을 생성
// .count() : 갯수 반환 (반환형 long)

File: 프로그래머스/0/120583. 중복된 숫자 개수/중복된 숫자 개수.py
def solution(array, n):
    return array.count(n)

# 다르 풀이 1
return sum(1 for x in array if x == n)
# 리스트 컴프리헨션: [표현식 for 항목 in iterable if 조건]
    # for x in array: array의 각 요소(x)에 대해 반복
    # if x == n : x가 n일 경우 
    # 1 : 표현식 수행 (1 생성)
# sum(): 생성된 1들의 합계 (즉, count 처럼 동작)

# 다른 풀이 2
answer = 0

for i in range(len(array)):
    if array[i] == n:
        answer += 1
        
return answer
